src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -397,6 +397,21 @@ internal Symbol BindNamespaceOrTypeOrAliasSymbol(ExpressionSyntax syntax, Diagno
 
         private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnostics)
         {
+            // If the tuple type is bound in a declaration context then we must have the
```

Comment:
At the moment, because empty names are recorded with `null`, the attribute is required. There is an optimization listed in https://github.com/dotnet/roslyn/issues/12347 that details how we could go without them, though, and once that change is made I see no reason why we couldn't relax this restriction.